src/rules/no-empty-fields.ts

```diff
@@ -0,0 +1,91 @@
+import { createRule } from "../createRule";
+import type { AST as JsonAST } from "jsonc-eslint-parser";
+
+import * as ESTree from "estree";
+
+export const rule = createRule({
+	create(context) {
+		return {
+			"Program > JSONExpressionStatement > JSONObjectExpression > JSONProperty[key.type=JSONLiteral]"(
+				node: JsonAST.JSONProperty & {
+					key: JsonAST.JSONStringLiteral;
+				},
+			) {
+				if (
+					!["JSONArrayExpression", "JSONObjectExpression"].includes(
+						node.value.type,
+					)
+				) {
+					return;
+				}
+
+				if (
+					(node.value.type === "JSONArrayExpression" &&
+						!node.value.elements.length) ||
+					(node.value.type === "JSONObjectExpression" &&
+						!node.value.properties.length)
+				) {
+					context.report({
+						data: { field: node.key.value },
+						messageId: "emptyFields",
+						node: node as unknown as ESTree.Node,
+						suggest: [
+							{
+								*fix(fixer) {
+									const keys = Object.keys(
```

Comment:
There shouldn't be a need to parse json here.  We already have the full AST available to us.  I see that you're parsing it so that you can get the index, which you then use to determine whether the comma from the previous line should be removed.  Rather than using the index to make that decision you could check if the tokenBefore is a comma (using `sourceCode.getTokenBefore`).  So the condition becomes "if the token before this node is a comma (i.e. it's not the first element) and the token after this node _is not_ a comma (i.e. it's the last element)", and that should be enough to determine whether to remove the tokenBefore.